jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -351,12 +351,27 @@ private static void arraycopy(Object[] A1, int offset1, Object[] A2, int offset2
 private static final int FileEncoding = 2;
 private static final int OSEncoding = 3;
 
+/*[IF OpenJ9-RawBuild]*/
+	/* This is a JCL native required only by OpenJ9 raw build.
+	 * OpenJ9 raw build is a combination of OpenJ9 and OpenJDK binaries without JCL patches within extension repo.
+	 * Currently OpenJ9 depends on a JCL patch to initialize platform encoding which is not available to raw build.
+	 * A workaround for raw build is to invoke this JCL native which initializes platform encoding.
+	 * This workaround can be removed if that JCL patch is not required.
+	 */
+private static native Properties initProperties(Properties props);
```

Comment:
Please investigate calling this native even for non-raw builds. We'll have to compile java.base/share/native/javalib/System.c instead of stubbing it out. What problems does this cause?

Reply:
As per Dan's comments earlier, calling this native performs unnecessary work which is not desirable for non-raw builds. Instead we choose to provide a `JCL` patch to export a previously internal API `InitializeEncoding` (please refer `Java 11` extension repo PR). 
We hope to contribute this patch back to `OpenJDK` eventually.